src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -513,9 +513,9 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
                 //       If we had to find one from a layer above that, parent->_MaskingActions
                 //       would have found it, so we inherit it for free!
                 const auto& inheritedCmd{ parent->_GetActionByID(actionID) };
-                if (inheritedCmd.has_value() && inheritedCmd.value())
+                if (inheritedCmd && *inheritedCmd)
                 {
-                    const auto& inheritedCmdImpl{ get_self<Command>(inheritedCmd.value()) };
+                    const auto& inheritedCmdImpl{ get_self<Command>(*inheritedCmd) };

```

Comment:
imo the original is clearer to read - is there a specific reason to do it this way?

Reply:
Yeah, the compiler can fail to remove the error checking and exception handling from a call to `.value()` even if you previously already checked whether `std::optional` contains something. The `operator*` will always return a direct zero-overhead reference into the optional's contents instead.

Additionally if we ever end up enabling clang-tidy in our project it'll nag for using `value()` instead of `operator*` in such situations. I think it's fairly likely that we'll use clang-tidy in the future and as such I think it's alright to change such case prematurely (if we're touching the code anyways).

In this case though, it's a function I didn't exactly change, so I can revert it if you want to.

Reply:
ok, good enough for me!